app/main/authorize/keycloak_login_required_decorator.py

```diff
@@ -80,3 +73,46 @@ def _check_if_user_has_access_to_ayr(keycloak_ayr_user_group, decoded_token):
         if keycloak_ayr_user_group in group:
             group_exists = True
     return group_exists
+
+
+def _decode_token(access_token):
+    keycloak_openid = keycloak.KeycloakOpenID(
+        server_url=current_app.config["KEYCLOAK_BASE_URI"],
+        client_id=current_app.config["KEYCLOAK_CLIENT_ID"],
+        realm_name=current_app.config["KEYCLOAK_REALM_NAME"],
+        client_secret_key=current_app.config["KEYCLOAK_CLIENT_SECRET"],
+    )
+
+    decoded_token = keycloak_openid.introspect(access_token)
+
+    return decoded_token
+
+
+def get_user_transferring_body_groups():
+    """
+    function that returns list of transferring body group names based on assignment of user group in keycloak for a user
+
+    User has user group assigned in keycloak, keycloak user groups are linked to transferring bodies
+    Based on active access token of user, user groups are extracted and list of user groups details retrieved
+
+    Args:
+        N/A
+
+    Returns:
+        function: return list of transferring bodies
+    """
+
+    access_token = session.get("access_token")
+    user_group_list = []
+    if access_token:
```

Comment:
Could we replace the check that uses `found_index = group.find("transferring_body")` 
 with `group.starts_with("transferring_body/") to ensure we are looking for `transferring_body/` at the start of the group?


Either way, it could be worth inverting the condition blocks here so that we aren't nesting so much if you agree that its easier to comprehend - might be a personal choice though

e.g.

```
    if not access_token:
        return []
   decoded_token = _decode_token(access_token)
   if not decoded_token["active"]:
       return []

   groups = decoded_token["groups"]
   
   users_transferring_bodies = []   
   for group in groups:
        if not group.starts_with("transferring_body/"):
            continue
        split_str = group.split("transferring_body/")
        if len(split_str) == 1:
           continue
        transferring_body = (split_str[2]).strip()
        users_transferring_bodies.append(transferring_body)
    return users_transferring_bodies
```

Reply:
created a new file : kecloak_manager.py and amended logic for get_user_transferring_body_groups() function